src/irr/asset/CImageLoaderJPG.cpp

```diff
@@ -285,19 +278,20 @@ asset::SAssetBundle CImageLoaderJPG::loadAsset(io::IReadFile* _file, const asset
 	uint32_t& height = imageSize[1];
 
 	// Allocate memory for buffer
-	output = new asset::ICPUBuffer(rowspan*height);
+	auto output = core::make_smart_refctd_ptr<asset::ICPUBuffer>(rowspan*height);
 
 	// Here we use the library's state variable cinfo.output_scanline as the
 	// loop counter, so that we don't have to keep track ourselves.
 	// Create array of row pointers for lib
-	rowPtr = new uint8_t* [height];
+	constexpr uint32_t MaxJPEGResolution = 65535u;
+	uint8_t* rowPtr[MaxJPEGResolution];
 	for (uint32_t i = 0; i < height; ++i)
 		rowPtr[i] = &reinterpret_cast<uint8_t*>(output->getPointer())[i*rowspan];
 
 	// Read rows from bottom order to match OpenGL coords
 	uint32_t rowsRead = 0;
 	while (cinfo.output_scanline < cinfo.output_height)
-		rowsRead += jpeg_read_scanlines(&cinfo, &rowPtr[cinfo.output_height - 1 - rowsRead], 1);
+		rowsRead += jpeg_read_scanlines(&cinfo, &rowPtr[rowsRead], 1);
```

Comment:
@Crisspl that was our problem XD

Reply:
yea there was some reason why someone did this but I don't remember

Reply:
after you get back BRDF Explorer needs a fix